src/bin/fp_read_unsigned_bin.c

```diff
@@ -55,7 +56,10 @@ void fp_read_unsigned_bin(fp_int *a, const unsigned char *b, int c)
   for (; c > 0; c--) {
      fp_mul_2d (a, 8, a);
      a->dp[0] |= *b++;
-     a->used += 1;
+
+     if (a->used == 0) {
```

Comment:
I'm not a fan of condition on every loop but when a->used is incremented on every iteration, the last iteration overruns the a->dp array in fp_mul_2d even when c is truncated properly.